impl/src/main/java/jakarta/faces/convert/BigDecimalConverter.java

```diff
@@ -72,51 +75,34 @@ public class BigDecimalConverter implements Converter {
      * @throws NullPointerException {@inheritDoc}
      */
     @Override
-    public Object getAsObject(FacesContext context, UIComponent component, String value) {
-
-        if (context == null || component == null) {
-            throw new NullPointerException();
-        }
+    public BigDecimal getAsObject(FacesContext context, UIComponent component, String value) {
+        notNullArgs( context , component );
 
-        // If the specified value is null or zero-length, return null
-        if (value == null) {
-            return null;
-        }
-        value = value.trim();
-        if (value.length() < 1) {
+        // If the specified value is null or blank, return null
+        if ( value == null || value.isBlank() ) {
             return null;
         }
 
         try {
-            return new BigDecimal(value);
+            return new BigDecimal(value.trim());
```

Comment:
Microoptimization comment: trim()+length() is more efficient than isBlank()+trim(). As performance is important this change should be reverted IMO. 

Try to make the PR as focused as possible as the ticket describes (introducing generics in faces converers).

Reply:
questionable... but ok ;)

probably it depends a lot on how many 
empty/blank values vs real values (with vs without empty chars...)
the webapp have to convert on the average

the `isBlank` algo is very fast and cheap 

the `trim` one involves 2 loops + clone operation